### PR TITLE
sign up, verify otp and login completed

### PR DESCRIPTION
**Pull request by Faisal-Sha**

### CODE REVIEW

File: src/SystemLayer/PersistenceDataLayer/DataAccessLayers/UserDAO.js [hunk 1]
```diff
@@ -1,19 +1,46 @@
-const Users = require('../DatabaseObjects/User')
-const senOtp = require('../../../Services/twilio')
+const Users = require('../DatabaseObjects/User');
+const senOtp = require('../../../Services/twilio');
```
Comment (Imrankhankhattar) [inline src/SystemLayer/PersistenceDataLayer/DataAccessLayers/UserDAO.js]: function name should be sendOtp

File: src/SystemLayer/PersistenceDataLayer/DataAccessLayers/UserDAO.js [hunk 2]
```diff
@@ -1,19 +1,46 @@
...
+        const OTP = generateOTP();
+        this.key = OTP.key;
+        // OTP sent to the user using twilio service
+        senOtp(data.contact, OTP);
+        // Store user data in Redis
+        setData(`user:${data.contact}`, data);
+        return { success: true, status: 200, message: "OTP sent!" }
+    }
+
+    async VerifyUser(data, login = false) {
```
Comment (Imrankhankhattar) [inline src/SystemLayer/PersistenceDataLayer/DataAccessLayers/UserDAO.js:22]: validate OTP method result is not being used?

**Review by Imrankhankhattar — CHANGES_REQUESTED** [review]

File: src/SystemLayer/PersistenceDataLayer/DataAccessLayers/UserDAO.js [hunk 3]
```diff
@@ -22,16 +22,17 @@ class UserDAO {
     async VerifyUser(data, login = false) {
         let user = await getData(`user:${data.contact}`);
         if (user) {
-            validateOTP(this.key, data.OTP);
-            if (login) {
+            const result = validateOTP(this.key, data.OTP);
```
Comment (Imrankhankhattar) [inline src/SystemLayer/PersistenceDataLayer/DataAccessLayers/UserDAO.js:25]: First OTP should be validated here, if it is matched then it should be verified further.

**Review by Imrankhankhattar — CHANGES_REQUESTED** [review]

File: src/SystemLayer/PersistenceDataLayer/DataAccessLayers/UserDAO.js (hunk 3 above)
Comment (Faisal-Sha) [reply src/SystemLayer/PersistenceDataLayer/DataAccessLayers/UserDAO.js:25]: can't get it

**Review by Imrankhankhattar — APPROVED** [review]

**Pull request merged**